File: application/smartgarden_flutter_socket/lib/src/models/sensor_reading.dart
class SensorReading {
  final int nodeId;
  final double? temperature;
  final double? humidity;
  final int? ldr;
  final int? soil;
  final DateTime createdAt;

  SensorReading({
    required this.nodeId,
    this.temperature,
    this.humidity,
    this.ldr,
    this.soil,
    required this.createdAt,
  });

  factory SensorReading.fromJson(Map<String, dynamic> j) => SensorReading(
        nodeId: (j['node_id'] as num).toInt(),
        temperature: (j['temperature'] as num?)?.toDouble(),
        humidity: (j['humidity'] as num?)?.toDouble(),
        ldr: (j['ldr'] as num?)?.toInt(),
        soil: (j['soil'] as num?)?.toInt(),
        createdAt: DateTime.parse(j['createdAt'] as String),
      );
}

File: server/src/v1/models/SensorReading.js
import mongoose from "mongoose";

const SensorReadingSchema = new mongoose.Schema(
  {
    node_id: { type: Number, required: true, index: true },
    temperature: { type: Number, required: false },
    humidity: { type: Number, required: false },
    ldr: { type: Number, required: false },
    soil: { type: Number, required: false },
    // raw payload if you want to keep it
    raw: { type: Object }
  },
  { timestamps: { createdAt: "createdAt", updatedAt: false } }
);

SensorReadingSchema.index({ node_id: 1, createdAt: -1 });

export default mongoose.model("SensorReading", SensorReadingSchema);
